src/Core/Managed/Shared/DataContracts/ExceptionTelemetry.cs

```diff
@@ -173,7 +189,43 @@ internal IList<ExceptionDetails> Exceptions
         {
             get { return this.Data.exceptions; }
         }
-        
+
+#if !NETSTANDARD1_3
+        /// <summary>
+        /// Set parsedStack from an array of StackFrame objects.
+        /// </summary>
+        public void SetParsedStack(System.Diagnostics.StackFrame[] frames)
+        {
+            List<StackFrame> orderedStackTrace = new List<StackFrame>();
+
+            if (this.Exceptions != null && this.Exceptions.Count > 0)
+            {
+                if (frames != null && frames.Length > 0)
+                {
+                    int stackLength = 0;
+
+                    this.Exceptions[0].parsedStack = new List<StackFrame>();
+                    this.Exceptions[0].hasFullStack = true;
+
+                    for (int level = 0; level < frames.Length; level++)
```

Comment:
do you need to do it synchronously? Does the exceptions stack trace parsed synchronously as well?